app/repositories/comments.py

```diff
@@ -0,0 +1,238 @@
+
+
+async def fetch_one(
+    target_id: int | None = None,
+    target_type: str | None = None,
+    userid: int | None = None,
+    colour: str | None = None,
+) -> Comment | None:
+    """Fetch a comment entry from the database."""
+    if target_id is None and target_type is None and userid is None and colour is None:
+        raise ValueError("Must provide at least one parameter.")
+
+    query = f"""
+        SELECT {READ_PARAMS}
+          FROM comments
+         WHERE target_id = COALESCE(:target_id, target_id)
+           AND target_type = COALESCE(:target_type, target_type)
+           AND userid = COALESCE(:userid, userid)
+           AND colour = COALESCE(:colour, colour)
+    """
+    params: dict[str, Any] = {
+        "target_id": target_id,
+        "target_type": target_type,
+        "userid": userid,
+        "colour": colour,
+    }
+    comment = await app.state.services.database.fetch_one(query, params)
+
+    return cast(Comment, dict(comment._mapping)) if comment is not None else None
+
+
+async def fetch_count(
+    target_id: int | None = None,
+    target_type: str | None = None,
+    userid: int | None = None,
+    time: float | None = None,
+    comment: str | None = None,
+    colour: str | None = None,
+) -> int:
+    """Fetch the number of comments in the database."""
+    query = """\
+        SELECT COUNT(*) AS count
+          FROM comments
+        WHERE target_id = COALESCE(:target_id, target_id)
+          AND target_type = COALESCE(:target_type, target_type)
+          AND userid = COALESCE(:userid, userid)
+          AND time = COALESCE(:time, time)
+          AND comment = COALESCE(:comment, comment)
+          AND colour = COALESCE(:colour, colour)
+    """
+    params: dict[str, Any] = {
+        "server": target_id,
+        "set_id": target_type,
+        "status": userid,
+        "artist": time,
+        "creator": comment,
+        "filename": colour,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    assert rec is not None
+    return cast(int, rec._mapping["count"])
+
+
+async def fetch_all(
+    score_id: int | None = None,
+    map_set_id: int | None = None,
+    map_id: int | None = None,
+) -> list[Comment]:
+    """Fetch a list of comments from the database."""
+    query = f"""\
+        SELECT c.time, c.target_type, c.colour, c.comment, u.priv
+          FROM comments c
+         INNER JOIN users u ON u.id = c.userid
+         WHERE (c.target_type = 'replay' AND c.target_id = :score_id)
+         OR (c.target_type = 'song' AND c.target_id = :map_set_id)
+         OR (c.target_type = 'map' AND c.target_id = :map_id)
+    """
+    params: dict[str, Any] = {
+        "score_id": score_id,
+        "map_set_id": map_set_id,
+        "map_id": map_id,
+    }
+
+    comments = await app.state.services.database.fetch_all(query, params)
+    return cast(list[Comment], [dict(c._mapping) for c in comments])
+
+
+async def update(
```

Comment:
i don't think we need update/delete for comments - should be create&read only i think